ts/src/connections/tunnelConnector.ts

```diff
@@ -12,5 +13,5 @@ export interface TunnelConnector {
      * @param isReconnect A value indicating if this is a reconnect (true) or regular connect (false).
      * @param cancellation Cancellation token.
      */
-    connectSession(isReconnect: boolean, cancellation: CancellationToken): Promise<void>;
+    connectSession(isReconnect: boolean, cancellation: CancellationToken, httpAgent?: http.Agent): Promise<void>;
```

Comment:
By convention, `cancellation` should be the final parameter. So insert `httpAgent` before it.

Reply:
Same in other places.

Reply:
When I attempt to do that it says that since `httpAgent` is an optional parameter it cannot be followed by required parameters which is why I put it after the cancellation token. The reason I didn't make it a required parameter was because I thought that there would be instances where you wouldn't have to pass the proxy in for connection.